p2p/p2p.go

```diff
@@ -1,7 +1,9 @@
 package p2p
 
 import (
+	"bytes"
 	"context"
+	"encoding/gob"
```

Comment:
right now we use `cbor` for that, do you think we can keep using it without relying on `gob` ?

Reply:
Didn't realize we were using cbor, it's a better option.